sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotCheckZeroSizeCollection.cs

```diff
@@ -31,55 +30,20 @@ namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
-    public sealed class DoNotCheckZeroSizeCollection : SonarDiagnosticAnalyzer
+    public sealed class DoNotCheckZeroSizeCollection : DoNotCheckZeroSizeCollectionBase<SyntaxKind, BinaryExpressionSyntax, ExpressionSyntax>
     {
-        internal const string DiagnosticId = "S3981";
-        private const string MessageFormat = "The {0} of '{1}' is always '>=0', so fix this test to get the real expected behavior.";
-
         private static readonly DiagnosticDescriptor rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+        protected override DiagnosticDescriptor Rule => rule;
+        protected sealed override Helpers.GeneratedCodeRecognizer GeneratedCodeRecognizer => Helpers.CSharp.GeneratedCodeRecognizer.Instance;
```

Comment:
I would move all expressions on the second line

Reply:
I think I prefer it this way. Yes, sometimes you have to scroll to see the full function, but the advantage is that it lines up much nicer. And with good naming, looking at the implementation detail shouldn't be needed ;)

Reply:
I would go +1 with @valhristov and move the expression (the part on the right side of the `=>` to the new line).

Reply:
I'll stubbornly persist on this in that case for simple accessors. @duncanp-sonar also likes this version more (or rather, dislikes it less than other variants).

Reply:
I have a slight preference for the single line version...